util/git/git_test.go

```diff
@@ -139,6 +140,37 @@ func TestLsRemote(t *testing.T) {
 	}
 }
 
+// Running this test requires git-lfs to be installed on your machine.
+func TestLFSClient(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "git-client-lfs-test-")
```

Comment:
Looks like we will need to install git-lfs in circle ci config:

https://discuss.circleci.com/t/is-git-lfs-supported-by-circleci-2-0/11283/7


```
          name: Install Git LFS
          command: |
            curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash
            apt-get update
            apt-get install -y git-lfs openssh-client
```

Reply:
Thanks a lot @alexmt - I got CircleCI to install & use the Git LFS plugin now. Unfortunately, this led to another error, which I believe is CircleCI specific. The LFS test fails with:

```
Error Trace:    git_test.go:162
Error:          Received unexpected error:
                `git lfs fetch --all` failed exit status 2: batch request: Invalid deploy key for argoproj-labs/argocd-testrepo-lfs: exit status 1
                error: failed to fetch some objects from 'https://github.com/argoproj-labs/argocd-testrepo-lfs.git/info/lfs'
Test:           TestLFSClient
```
I changed test repositories, but to no avail A quick search on Google revealed not much to this specific error, and all I found seems to be CircleCI related. The LFS works from multiple machines I tested on, just not in the CircleCI build. I will look through their support base for possible solutions, or write to their support.

Reply:
We are going to dockerize tests since it's been pretty painful to configure Circle CI same as real production environment. I would suggest moving your test to `test/e2e` directory and skip it during CI run using `test.LocalOnly(t)` ([example](https://github.com/argoproj/argo-cd/blob/9a76a06f3962951dba483c1670b643fd32ee1a3c/test/e2e/app_of_apps_test.go#L13))

After dockerizing the tests we will enable this test in CI

Reply:
I have found the reason and could fix it. CircleCI injects a SSH key to the user's $HOME/.ssh, which is reason for LFS failing (doesn't like deploy keys). Actually, I guess that was one of the reasons why HOME was set to /dev/null in Git client implementation -- and I removed that with one of the previous commits. It's back in place now, LFS now also works in the CI environment.